lib/RGBot.js

```diff
@@ -583,196 +1098,360 @@ const RGBot = class {
+     *         if (candidateBlock) {
+     *             if (await bot.approachAndDigBlock(candidateBlock)) {
+     *                 // you can collect the item here, but ideally you would let the collection decision be part of your main loop next pass
+     *                 // optionally wait a few ticks for the item to spawn, but that is wasting time that could be used making another action/decision
+     *                 const theItem = this.findItemsOnGroundv2({itemNames: ['name1',...], maxDistance: 5}).shift()?.result
+     *                 if (theItem) {
+     *                     return await this.collectItemOnGround(theItem)
+     *                 }
+     *             }
+     *         }
      */
     async findAndDigBlock(blockType, options = {}) {
-        const partialMatch = options.partialMatch || false;
-        const onlyFindTopBlocks = options.onlyFindTopBlocks || false;
-        const maxDistance = options.maxDistance || 50;
-        const skipCollection = options.skipCollection || false;
-        const skipClosest = options.skipClosest || false;
-
-        let result = false;
-        const block = this.findBlock(blockType, { partialMatch, onlyFindTopBlocks, maxDistance, skipClosest });
-        if (block) {
-            if(skipCollection) {
-                // if we don't care about drops,
-                // then return true to indicate that the Block was broken
-                result = true
-            }
-            else
-            {
-                // if we care about drop then try to identify whether
-                // anything has dropped from breaking the Block
-                // and make the Bot move to collect it
-                try {
-                    if (await this.approachBlock(block)) {
-                        await this.digBlock(block);
-
-                        let droppedItem = null;
-                        await this.wait(25); // give the server time to create drops
-                        if(!block.drops || block.drops.length === 0) {
-                            droppedItem = this.findItemOnGround(block.name);
-                        }  else {
-                            const itemDefinition = this.getItemDefinitionById(block.drops[0].drop.id);
-                            if(!!itemDefinition) {
-                                const itemName = this.getEntityName(itemDefinition)
-                                droppedItem = !!itemName ? this.findItemOnGround(itemName) : null;
-                            }
-                        }
-                        if (droppedItem) {
-                            await this.approachEntity(droppedItem);
-                        }
-                        result = true;
+        const skipCollection = options.skipCollection || false
+        const maxDistance = options.maxDistance || 50
+        this.#logWarn(`findAndDigBlock(blockType, options): Deprecated since 1.2.0, please see doc example for new code`)
```

Comment:
I like this comment, good to see the player explicitly notified in their logs of using deprecated code.